astropy/modeling/core.py

```diff
@@ -2015,6 +2015,12 @@ class members that should be added to the generated
             instance._user_inverse = mcls._make_user_inverse(
                     operator, left, right)
 
+            if left._n_models == right._n_models:
+                instance._n_models = left._n_models
+            else:
+                raise ValueError('Model sets must have the same number of '
+                                 'components.')
```

Comment:
The error message can be more informative. Perhaps listing the conflicting `_n_models` values.